Game/space-invaders/src/enemies/EnemyManager.h

```diff
@@ -15,13 +15,36 @@ class EnemyManager
     
     void Update(float deltaTime);
     void Render(const SpriteRenderer& renderer);
-    void CheckCollisions(GameObject& projectile);
+    void HandleEnemyHit(GameObject& enemy);
+    bool IsEveryEnemyKilled() const;
 
     std::vector<GameObject>& GetEnemies() { return m_Enemies; }
     
 private:
 
-    const float SHOOT_SECONDS_COOLDOWN = 1.5f;
+    static constexpr unsigned int TOTAL_DIFFICULTY_CURVE_POINTS = 5;
```

Comment:
Is there a reason for the const be static?

Reply:
It needs to be static constexpr so we can use this number as the size of the array. It will turn into a compile time value and it will be accepted as size of const arrays, otherwise it wouldn't be.